docs/concepts/linq.rst

```diff
@@ -0,0 +1,268 @@
+
+Expressive Code is Valuable
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ask a Ruby programmer why they like Ruby.  Their answer will probably involve something along the lines of "I can express my intent directly in my code".  The same can be said for LINQ.
+
+Imagine you have a list of pets, but want to convert it into a lookup where you can access a pet directly by its ``RFID`` value.
+
+Traditional imperative code:
+
+.. code-block:: c#
+
+	var petLookup = new Dictionary<int, Pet>();
+	
+	foreach (var pet in pets)
+	{
+	    petLookup.Add(pet.RFID, pet);
+	}
+
+Equivalent LINQ expression:
+
+.. code-block:: c#
+
+	var petLookup = pets.ToDictionary(pet => pet.RFIDValue, pet => pet);
+	
+The intention behind the code is not to create a new ``Dictionary<int, Pet>`` and add to it via a loop, it is to convert an existing list into a lookup!  LINQ preserves these semantics whereas the imerative code does not.
+
+Expressive code is valuable because it evens the playing field between intent and code when reasoning as a programmer.  LINQ accomplishes this.
+
+Another bonus is code brevity.  Imagine reducing large portions of a codebase by 1/3 as done above.  Pretty sweet deal, right?
+
+LINQ Providers
+^^^^^^^^^^^^^^
+
+For a significant chunk of software out in the wild, everything revolves around dealing with data from some source (Databases, JSON, XML, etc).  Most APIs which do this can be a pain to work with.  Enter LINQ Providers.
+
+Consider the following: finding all XML elements with a specific attribute value.
+
+.. code-block:: c#
+
+	public static FindAllElementsWithAttribute(XElement documentRoot, string elementName,
+	                                           string attributeName, string value)
+	{
+	    return from el in documentRoot.Elements(elementName)
+	           where (string)el.Element(attributeName) == value
+	           select el;
+	}
+	
+Writing code to manually traverse the XML document to perform this task would be pretty ugly.
+
+Interacting with XML isn't the only thing you can do with LINQ Providers.  `Linq to SQL <https://msdn.microsoft.com/en-us/library/bb386976(v=vs.110).aspx>`_ is a fairly bare-bones ORM for an MSSQL Server Database.  The `JSON.NET <http://www.newtonsoft.com/json/help/html/LINQtoJSON.htm>`_ library provides efficient JSON Document traversal via LINQ.  Furthermore, if there isn't a library which does what you need, you can also `write your own LINQ Provider <https://msdn.microsoft.com/en-us/library/vstudio/Bb546158(v=VS.110).aspx>`_!  
+
+Essential Samples
+-----------------
+
+For a truly comprehensive list of LINQ samples, visit `101 LINQ Samples <https://code.msdn.microsoft.com/101-LINQ-Samples-3fb9811b>`_.
+
+The following is a quick demonstration of some of the essential pieces of LINQ.  This is in no way comprehensive, as LINQ provides significantly more functionality than what is showcased here.
+
+* The bread and butter - ``Where``, ``Select``, and ``Aggregate``.
```

Comment:
Would be good to liberally sprinkle links to 101 linq samples.

Should we move those samples to GitHub?